integration_tests/ethereum/contracts/MockAaveV2StablecoinCellar.sol

```diff
@@ -5,109 +5,129 @@ import "./interfaces.sol";
+    event mockReinvest(uint256 minAssetsOut);
+    event mockClaimAndUnstake();
+    event mockSweep(address token, address to);
+    event mockSetLiquidityLimit(uint256 limit);
+    event mockSetDepositLimit(uint256 limit);
+    event mockSetTrust(address position, bool trust);
+    event mockSetShutdown(bool shutdown, bool existPosition);
+
+    constructor() {}
 
     /**
      * @notice Take platform fees and performance fees off of cellar's active assets.
      */
-    function accrueFees() external onlyOwner {
+    function accrueFees() external {
         emit mockAccrueFees();
     }
 
-    event mockTransferFees();
-
     /**
-     * @notice Transfer accrued fees to Cosmos to distribute.
+     * @notice Transfer accrued fees to the Sommelier Chain to distribute.
      */
     function transferFees() external onlyOwner {
         emit mockTransferFees();
     }
 
-    event mockEnterStrategy();
+    /**
+     * @notice Update the address of the fee distributor on the Sommelier Chain. IMPORTANT: Ensure
+     *         that the address is formatted in the specific way that the Gravity contract expects
+     *         it to be.
+     */
+    function setFeesDistributor(bytes32 newFeesDistributor) external onlyOwner {
+        emit mockSetFeesDistributor(newFeesDistributor);
+    }
 
     /**
      * @notice Enters into the current Aave stablecoin strategy.
      */
-    function enterStrategy() external onlyOwner  {
-        emit mockEnterStrategy();
+    function enterPosition() external onlyOwner  {
+        emit mockEnterPosition();
     }
 
-    event mockRebalance(address[] path, uint256 amountOutMinimum);
-
     /**
-     * @notice Rebalances current assets into a new asset strategy.
-     * @param path path to swap from the current asset to new asset using Uniswap
-     * @param amountOutMinimum minimum amount of assets returned after swap
+     * @notice Rebalances current assets into a new asset position.
+     * @param route array of [initial token, pool, token, pool, token, ...] that specifies the swap route
+     * @param swapParams multidimensional array of [i, j, swap type] where i and j are the correct
+                         values for the n'th pool in `_route` and swap type should be 1 for a
+                         stableswap `exchange`, 2 for stableswap `exchange_underlying`, 3 for a
+                         cryptoswap `exchange`, 4 for a cryptoswap `exchange_underlying` and 5 for
+                         Polygon factory metapools `exchange_underlying`
+     * @param minAssetsOut minimum amount of assets received from swap
      */
-    function rebalance(address[] memory path, uint256 amountOutMinimum) external onlyOwner  {
-        emit mockRebalance(path, amountOutMinimum);
+    function rebalance(
+        address[9] memory route,
+        uint256[3][4] memory swapParams,
+        uint256 minAssetsOut
+    ) external onlyOwner  {
```

Comment:
Does that give us any more info that validating in steward does?

Reply:
The integration test is meant to be testing whether or not the calls generated by steward are hitting the contract correctly. We're only using a mock contract because using the real one is infeasible.